papis/commands/addto.py

```diff
@@ -57,35 +57,63 @@ def run(document: papis.document.Document,
         raise DocumentFolderNotFound(papis.document.describe(document))
 
     import shutil
-    from papis.paths import symlink, rename_document_files
+    from papis.paths import (symlink, rename_document_files,
+                             download_remote_files, is_remote_file)
+
+    # ensure all remote files are downloaded, before we start renaming.
+    # the number of
+    local_files = download_remote_files(filepaths_arg)
+
+    # we only symlink a file it isn't a downloaded file that is stored in
+    # a temporary directory. We keep track of a boolean per file that
+    # tells us whether to link
+    filepaths: List[Tuple[str, bool]] = []
+    for orig_filepath, filepath in zip(filepaths_arg, local_files):
+        # skip remote files that haven't been properly downloaded
+        if not filepath:
+            continue
+
+        if os.path.exists(filepath):
+            filepaths.append((filepath,
+                              link if not is_remote_file(orig_filepath) else False))
+        else:
+            logger.warning("Skipping non-existent file: %r", filepath)
 
-    new_file_list = rename_document_files(
-        document, filepaths, file_name_format=file_name
+    if not filepaths:
+        logger.error("No valid files provided")
+        return
+
+    # new_filenames is a list of renamed filenames. rename_document_files ensures
+    # that here is no name collision even after renaming so we can be sure that
+    # there is a unique filename for every file in the filepaths list.
+    new_filenames = rename_document_files(
+        document, [f[0] for f in filepaths], file_name_format=file_name
     )
 
-    for in_file_path, out_file_name in zip(filepaths, new_file_list):
+    for (in_file_path, link_file), out_file_name in zip(filepaths, new_filenames):
+
         out_file_path = os.path.join(doc_folder, out_file_name)
         if os.path.exists(out_file_path):
             logger.warning("File '%s' already exists. Skipping...", out_file_path)
             continue
 
-        if link:
-            logger.info("[LN] '%s' to '%s'.", in_file_path, out_file_name)
+        if link_file:
+            logger.debug("[LN] '%s' to '%s'.", in_file_path, out_file_name)
```

Comment:
Why make these into debug logs? It seemed generally useful to see what the files got renamed to.

Reply:
It seemed unnecessarily verbose to me and maybe an oversight from when this feature came in. Maybe it is just me personally, but I don't want to see this. I just want papis to do the thing I configured it to.

Reply:
Fair enough, it's tricky to show just the right amount™ of information here. For the moment, this is inconsistent with `papis add` (which also has the same info logs) and a bit out of scope, so maybe leave it for another PR?